src/app/debug/debug-tree/debug-tree.component.ts

```diff
@@ -129,94 +119,74 @@ export class DebugTreeComponent implements AfterViewInit, OnDestroy {
+      this.tree.clearItems();
     }
-
-    this.treeReference.addTo(tree, null);
-    this.treeReference.selectItem(
-      // @ts-ignore
-      this.treeReference.getItems()[
-        this.treeReference.getItems().findIndex((item: any) => item.id == tree.items[0].id)
-      ],
-    );
+    const newReport: CreateTreeItem = this.transformReportToHierarchyStructure(report);
+    const optional: OptionalParameters = { childrenKey: 'checkpoints', pathAttribute: 'uid' };
+    const path: string = this.tree.addItem(newReport, optional);
+    this.tree.selectItem(path);
     this.hideOrShowCheckpointsBasedOnView(this.currentView);
   }
 
-  selectReport(event: any): void {
-    this.selectReportEvent.emit(event);
-  }
-
-  removeReport(report: any): void {
-    let parentItem: any = this.findParentNode(this.treeReference.getItems().find((item: any) => item.value == report));
-    let root = parentItem.element.parentNode;
-    this.treeReference.removeItem(parentItem);
-    parentItem.element.remove();
-
-    let latestAddedReport = root.lastChild;
-    if (latestAddedReport) {
-      this.treeReference.selectItem(latestAddedReport.querySelectorAll('li')[0]);
+  //Ladybug reports don't have a parent-child structure for its checkpoints, this function creates that parent-child structure
+  transformReportToHierarchyStructure(report: Report): Report {
+    const checkpoints = report.checkpoints;
+    for (let i = 0; i < checkpoints.length; i++) {
+      checkpoints[i].icon = this.helperService.getImage(
+        checkpoints[i].type,
+        checkpoints[i].encoding,
+        checkpoints[i].level,
+      );
+      if (i - 1 >= 0 && checkpoints[i].type > 1) {
+        if (!checkpoints[i - 1].checkpoints) {
+          checkpoints[i - 1].checkpoints = [];
+        }
+        checkpoints[i - 1].checkpoints!.push(checkpoints[i]);
+        checkpoints.splice(i, 1);
+        i--;
+      }
     }
+    report.checkpoints = checkpoints;
+    return report;
   }
 
-  findParentNode(item: any): any {
-    if (item.prevItem != undefined && item.parentElement) {
-      return this.findParentNode(item.prevItem);
-    }
+  selectReport(value: FileTreeItem): void {
+    this.selectReportEvent.emit(value.originalValue);
+  }
 
-    return item;
+  removeReport(report: any): void {
+    this.tree.removeItem(report.path);
   }
 
   closeEntireTree(): void {
     this.closeEntireTreeEvent.emit();
-    this.treeReference.clear();
+    this.tree.clearItems();
   }
 
   expandAll(): void {
-    this.treeReference.expandAll();
+    this.tree.expandAll();
   }
 
   collapseAll(): void {
-    this.treeReference.collapseAll();
+    this.tree.collapseAll();
   }
 
   downloadReports(exportBinary: boolean, exportXML: boolean): void {
     let queryString = '';
-    this.getTreeReports().forEach((report) => {
-      queryString += 'id=' + report.value.storageId + '&';
-    });
+    for (let treeReport of this.getTreeReports()) {
+      queryString += 'id=' + treeReport.storageId + '&';
```

Comment:
I know you didn't do this but:
```suggestion
      queryString += `id=${treeReport.storageId}&`;
```